src/components/OSCALEditableTextField.test.js

```diff
@@ -0,0 +1,151 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import OSCALEditableTextField from "./OSCALEditableTextField";
+import { testModifiableMetadata } from "../test-data/MetadataData";
+
+test("OSCALEditableTextField loads", () => {
+  render(
+    <OSCALEditableTextField
+      canEdit
+      editedField={["version"]}
+      isInEditState={[false, () => {}]}
+      modifiableData={testModifiableMetadata.version}
+    />
+  );
+
+  render(
+    <OSCALEditableTextField
+      canEdit
+      editedField={["version"]}
+      isInEditState={[true, () => {}]}
+      modifiableData={testModifiableMetadata.version}
+    />
+  );
+});
+
+export default function testOSCALEditableTextField(
+  parentElementName,
+  renderer
+) {
+  test(`${parentElementName} displays text field and can be edited`, () => {
+    renderer();
+
+    screen
+        .getByRole("button", {
+            name: "edit-system-security-plan-metadata-title",
+        })
+        .click();
+
+    const textField = screen.getByTestId(
+      "textField-system-security-plan-metadata-title"
+    );
+    expect(textField).toBeVisible();
+  });
+
+  test(`${parentElementName} tests if editing ends after 'Esc' key pressed`, () => {
+    renderer();
+
+    screen
+        .getByRole("button", {
+            name: "edit-system-security-plan-metadata-title",
+        })
+        .click();
+
+    const textField = screen.getByTestId(
+      "textField-system-security-plan-metadata-title"
+    );
+
+    fireEvent.keyDown(textField, {
+      key: 'Escape',
+      charCode: 27,
+    });
+
+    const editButton = screen.getByRole("button", {
+        name: "edit-system-security-plan-metadata-title",
+      });
+    expect(editButton).toBeVisible();
+  });
+
+  /* test(`${parentElementName} tests if text value remains same after 'Esc' key pressed`, () => {
+    renderer();
+
+    screen
+        .getByRole("button", {
+            name: "edit-system-security-plan-metadata-title",
+        })
+        .click();
+
+    const textField = screen.getByTestId(
+      "textField-system-security-plan-metadata-title"
+    );
+
+    fireEvent.keyDown(textField, {
+      key: "Escape",
+      code: "Escape",
+      keyCode: 27,
+      charCode: 27,
+    });
+
+    screen
+        .getByRole("button", {
+            name: "edit-system-security-plan-metadata-title",
+        })
+        .click();
+
+    const textFieldAfterEvent = screen.getByTestId(
+        "textField-system-security-plan-metadata-title"
+    );
+
+    expect(textFieldAfterEvent).toBe(textField);
+  }); */
+
+  test(`${parentElementName} tests if editing ends after 'Enter' key is pressed`, () => {
```

Comment:
We call this test from `OSCALSsp.test.js`, where we pass in a renderer, `sspRendererRestMode`. This renderer does not have the `onFieldSave` function passed into the Ssp component that is rendered. This is important because when an enter key is pressed, we attempt to save what is in the text field. With no `onFieldSave` function, React doesn't know what to do and will automatically lead to failures.

You could fix this by adding a dummy `onFieldSave={()=>{}}` function to `sspRendererRestMode`.